api/s2n.h

```diff
@@ -2291,8 +2291,23 @@ S2N_API extern int s2n_shutdown_send(struct s2n_connection *conn, s2n_blocked_st
 /**
  * Used to declare what type of client certificate authentication to use.
  *
- * Currently the default for s2n-tls is for neither the server side or the client side to use Client (aka Mutual) authentication.
- */
+ * A s2n_connection will enforce client certificate authentication (mTLS) differently based on
+ * the s2n_cert_auth_type and s2n_mode(client/server) of the connection, as described below. The
+ * **default** behavior is used if the application hasn't set an override via
+ * `s2n_config_set_client_auth_type()` or `s2n_connection_set_client_auth_type()`.
+ *
+ * **Server behavior:**
+ * - None(**default**): don't request client authentication.
+ * - Optional: request the client's certificate and validate if it's non-empty.
+ * - Required: request the client's certificate and validate it.
+ *
+ * **Client behavior:**
+ * - None: abort the handshake if the server requests client authentication.
+ * - Optional(**default**): send the client's certificate (can be empty if not set) if the
```

Comment:
Can you be explicit about how if the user doesn't provide a certificate no certificate will be sent?